### PR TITLE
2960 lesmis gross intake ratio

### PR DESCRIPTION
**Pull request by EMcQ-BES**
### Issue #: https://github.com/beyondessential/tupaia-backlog/issues/2960

### CODE REVIEW

**Review by edmofro — APPROVED** [review]
Amazing migration!! All looks awesome to me, well laid out, clever code reuse, and nice indicators.

Could you please add a summary of all of the indicators and their formulae to the ticket, so that Juliana and Erin know what exist and we can work off the same ones/make consistent others in future?

Also interested in how fast the dashboard item is to load? Nested indicators is still a performance bottleneck for us

File: packages/database/src/migrations/20210623220403-LESMISGrossIntakeRate-modifies-data.js
Comment (edmofro) [inline packages/database/src/migrations/20210623220403-LESMISGrossIntakeRate-modifies-data.js:52]: Please add a sort here so that Total is last (doesn't especially matter which of male/female is first)

File: packages/database/src/migrations/20210623220403-LESMISGrossIntakeRate-modifies-data.js
Comment (edmofro) [inline packages/database/src/migrations/20210623220403-LESMISGrossIntakeRate-modifies-data.js:87]: Please add a sort here so that Total is last (previous comment was in the wrong spot)

**Comment by edmofro** [comment]
@EMcQ-BES please update this one to have `valueType: 'percentage'`, and divide all values by 100

**Comment by edmofro** [comment]
@EMcQ-BES also https://github.com/beyondessential/tupaia-backlog/issues/2960#issuecomment-868171158

File: packages/database/src/migrations/20210623220403-LESMISGrossIntakeRate-modifies-data.js
Comment (edmofro) [inline packages/database/src/migrations/20210623220403-LESMISGrossIntakeRate-modifies-data.js:79]: @rohan-bes we should probably turn this into an internal `divide` function that gets attached to mathjs right?

**Comment by EMcQ-BES** [comment]
@edmofro Are you happy with me leaving the indicator stuff in here so we can come back to it? Or prefer I strip it out completely?

**Comment by edmofro** [comment]
Erm probably strip it out before we merge into the 2963 branch (the config should work with just the final aggregate data element codes right?), but keep it safe so you can come back to it on say Wednesday

File: packages/database/src/migrations/20210623220403-LESMISGrossIntakeRate-modifies-data.js
Comment (rohan-bes) [reply packages/database/src/migrations/20210623220403-LESMISGrossIntakeRate-modifies-data.js:79]: Such a function does actually already exist in this PR: https://github.com/beyondessential/tupaia/pull/2624
I've been horrible at getting that PR merged as it failed testing a little while back and I haven't got around to investigating the failure as yet, I'll prioritise it ASAP

Comment (rohan-bes) [reply packages/database/src/migrations/20210623220403-LESMISGrossIntakeRate-modifies-data.js:79]: Alternatively, if @EMcQ-BES wants to cherry pick 4e2b0f46c733c8c47a53b20302d10fa8ff578d03 and f1a807eb4d5d909467bc546f6c646f8674f1433c then we can work out a merging strategy based on whoever merges first?

Comment (edmofro) [reply packages/database/src/migrations/20210623220403-LESMISGrossIntakeRate-modifies-data.js:79]: @EMcQ-BES merge this now, and then on Wednesday when we return to it we can see if @rohan-bes's work has made it in to dev and either use that or cherry pick it at that point

**Review by edmofro — CHANGES_REQUESTED** [review]
Oh sorry still need to update to being a time series 🙏

**Review by edmofro — APPROVED** [review]
Thanks for keeping those last three commits clean, reviewing one by one was much easier

lgtm 🎉

**Pull request merged**
